src/components/Tool/Appeal/Modals/AddExcludedContactModal/AddExcludedContactModal.test.tsx

```diff
@@ -0,0 +1,209 @@
+            }>
+              mocks={{
+                Appeal: appealMock,
+              }}
+              onCall={mutationSpy}
+            >
+              <AppealsWrapper>
+                <AppealsContext.Provider
+                  value={{
+                    accountListId,
+                    appealId: appealId,
+                    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+                    // @ts-ignore
+                    contactsQueryResult: { refetch },
+                  }}
+                >
+                  <AddExcludedContactModal
+                    handleClose={handleClose}
+                    contactIds={contactIds}
+                  />
+                </AppealsContext.Provider>
+              </AppealsWrapper>
+            </GqlMockedProvider>
+          </TestRouter>
+        </ThemeProvider>
+      </SnackbarProvider>
+    </LocalizationProvider>
+  </I18nextProvider>
+);
+
+describe('AddExcludedContactModal', () => {
+  beforeEach(() => {
+    handleClose.mockClear();
+    refetch.mockClear();
+  });
+  it('default', () => {
+    const { getByRole, getByText } = render(<Components />);
+
+    expect(getByRole('heading', { name: 'Add Contact' })).toBeInTheDocument();
+    expect(
+      getByText(
+        'This contact has been previously excluded from this appeal. Are you certain you wish to add them?',
+      ),
+    ).toBeInTheDocument();
+    expect(getByRole('button', { name: 'No' })).toBeInTheDocument();
+    expect(getByRole('button', { name: 'Yes' })).toBeInTheDocument();
+  });
+
+  it('bulk default', () => {
+    const { getByRole, getByText } = render(
+      <Components contactIds={bulkContactIds} />,
+    );
+
+    expect(getByRole('heading', { name: 'Add Contacts' })).toBeInTheDocument();
+    expect(
+      getByText(
+        'These 3 contacts have been previously excluded from this appeal. Are you certain you wish to add them?',
+      ),
+    ).toBeInTheDocument();
+  });
+
+  it('should close modal', async () => {
+    const { getByRole } = render(<Components />);
+
+    expect(handleClose).toHaveBeenCalledTimes(0);
+    await waitFor(() =>
+      expect(getByRole('button', { name: 'No' })).not.toBeDisabled(),
+    );
+    userEvent.click(getByRole('button', { name: 'No' }));
+    expect(handleClose).toHaveBeenCalledTimes(1);
+
+    userEvent.click(getByRole('button', { name: 'Close' }));
+    expect(handleClose).toHaveBeenCalledTimes(2);
+  });
+
+  it('should send mutation and remove the contact from the excluded list', async () => {
+    const { getByRole } = render(<Components />);
+
+    expect(mutationSpy).toHaveBeenCalledTimes(0);
+    expect(refetch).not.toHaveBeenCalled();
+
+    await waitFor(() =>
+      expect(getByRole('button', { name: 'Yes' })).not.toBeDisabled(),
+    );
+    userEvent.click(getByRole('button', { name: 'Yes' }));
+
+    await waitFor(() => {
+      expect(mutationSpy.mock.calls[5][0].operation.operationName).toEqual(
```

Comment:
This is a good candidate for `expect().toHaveGraphqlOperation`. And the test below.